### PR TITLE
Task3 - Build a containerized FastAPI app 

### LINKED ISSUE #5: Add a containerized HelloWorld app using FastAPI and test it locally

### PR DESCRIPTION
**Pull request by lukyrasocha**
Contains also a short tutorial on how to rerun this experiment

### CODE REVIEW

**Comment by halvgaard** [comment]
Make sure the docker file commands are optimized for fast docker build caching. Order of commands is not optimal right now. Also pip has a flag to avoid caching (no need inside a onetime container) inside a container. 
But can't you install using poetry instead of pip?

**Comment by lukyrasocha** [comment]
#12

File: Task6/helloworld/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[tool.poetry]
+name = "helloworld"
+version = "0.1.0"
+description = ""
+authors = ["Lukas Rasocha <rasocha.lukas@gmail.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.9"
```
Comment (halvgaard) [inline Task6/helloworld/pyproject.toml]: python 3.9?

**Review by halvgaard — CHANGES_REQUESTED** [review]
Why is src outside the helloworld module? 
Do people still use src as a name in python (poetry) packages - what is best practice? 
ReadMe must have run and setup instructions.

File: Task6/helloworld/pyproject.toml (hunk 1 above)
Comment (lukyrasocha) [reply Task6/helloworld/pyproject.toml]: Fixed to 3.7 now

**Comment by lukyrasocha** [comment]
Updated everything mentioned in your previous comment.

File: Task6/helloworld/tests/test_helloworld.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from helloworld import __version__
+
```
Comment (halvgaard) [inline Task6/helloworld/tests/test_helloworld.py:4]: We need a unit test of the app method. Hint: use TestClient https://fastapi.tiangolo.com/tutorial/testing/

File: Task6/helloworld/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM python:3
+
+ENV PIP_NO_CACHE_DIR=1
+RUN pip3 install "poetry==1.1.12"
+
+RUN python -m venv /venv
```
Comment (halvgaard) [inline Task6/helloworld/Dockerfile]: Why do you need a separate virtual env inside the container?

**Review by halvgaard — CHANGES_REQUESTED** [review]

**Comment by lukyrasocha** [comment]
Updated based on the last comment

**Review by halvgaard — APPROVED** [review]

**Pull request merged**